python/sdss_access/sync/curl.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+# The line above will help with 2to3 support.
+
+from os.path import isfile, exists, dirname, join, basename, getsize, getmtime, sep
+from re import search
+from sdss_access import SDSSPath, AccessError
+from sdss_access.sync.auth import Auth
+from sdss_access.sync.stream import Stream
+import re
+from sdss_access import os_windows
+from os import popen
+from numpy import transpose
+from datetime import datetime, timedelta
+import time
+from pytz import timezone
+#import urllib
+try:
+    from urllib2 import HTTPPasswordMgrWithDefaultRealm, HTTPBasicAuthHandler, build_opener, install_opener, urlopen
+    from urllib2 import HTTPError
+except:
+    from urllib.request import HTTPPasswordMgrWithDefaultRealm, HTTPBasicAuthHandler, build_opener, install_opener, urlopen
+    from urllib.error import HTTPError
+
+
+class CurlAccess(SDSSPath):
```

Comment:
It looks like the majority of CurlAccess is exactly the same code as in RsyncAccess.  In which case, I think it's better to move all the common code into a base class, e.g. BaseAccess, and have RsyncAccess and CurlAccess subclass from that.   This will help readability and maintainability when we need to make changes to the underlying code.

Reply:
Agreed!  We shall factor out the base class.

Reply:
This looks good to me.